pkg/deploy/devfile_registry.go

```diff
@@ -151,6 +155,8 @@ func SyncDevfileRegistryToCluster(deployContext *DeployContext, cheHost string)
 			}
 		}
 
+		deployContext.InternalService.DevfileRegistryHost = fmt.Sprintf("%s://%s.%s.svc.cluster.local:%d", DevfileRegistryServiceProtocol, DevfileRegistry, deployContext.CheCluster.Namespace, DevfileRegistryServicePort)
```

Comment:
The internal URL of a service is `serviceName.namespace.svc.cluster.local` where everything apart of `serviceName` is optional. The `namespace` portion resolves to the current namespace of the "caller" if I am not mistaken, so this URL is identical to the above defined `routePathPrefix` (which is not a prefix but a URL :) ).

Therefore we can use just 1 of the two for these both purposes IMHO.

Reply:
Code was very changed after rebase with master.

Reply:
> The internal URL of a service is serviceName.namespace.svc.cluster.local where everything apart of serviceName is optional.

@metlos As far I remember url with `serviceName` will be working inside one namespace. `serviceName.namespace.svc.cluster.local` should work between namespaces inside cluster.